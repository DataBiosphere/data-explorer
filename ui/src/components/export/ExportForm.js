import React from "react";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogContentText from "@material-ui/core/DialogContentText";
import DialogTitle from "@material-ui/core/DialogTitle";

class ExportForm extends React.Component {
  render() {
    return (
      <Dialog
        open={this.props.open}
        onClose={this.props.handleClose}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title">Export to FireCloud</DialogTitle>
        <DialogContent>
          <DialogContentText>
            Enter a workspace to continue in a FireCloud notebook.
          </DialogContentText>
          <TextField
            autoFocus
            margin="dense"
            label="Workspace Namespace"
            fullWidth
          />
          <TextField margin="dense" label="Workspace Name" fullWidth />
        </DialogContent>
        <DialogActions>
          <Button onClick={this.props.handleClose} color="primary">
            Cancel
          </Button>
          <Button color="primary" disabled>
            Export
          </Button>
        </DialogActions>
      </Dialog>
    );
  }
}

export default ExportForm;
