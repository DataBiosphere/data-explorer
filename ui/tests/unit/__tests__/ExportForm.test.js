import React from "react";
import ExportForm from "../../../src/components/export/ExportForm";

test("Renders correctly", () => {
  const tree = shallow(<ExportForm />);
  expect(tree).toMatchSnapshot();
});
