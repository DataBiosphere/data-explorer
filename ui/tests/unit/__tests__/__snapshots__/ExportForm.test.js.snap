// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<WithStyles(Dialog)
  aria-labelledby="form-dialog-title"
>
  <WithStyles(DialogTitle)
    id="form-dialog-title"
  >
    Export to FireCloud
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(DialogContentText)>
      Enter a workspace to continue in a FireCloud notebook.
    </WithStyles(DialogContentText)>
    <TextField
      autoFocus={true}
      fullWidth={true}
      id="name"
      label="Workspace Namespace"
      margin="dense"
      required={false}
      select={false}
      type="email"
    />
    <TextField
      fullWidth={true}
      id="name"
      label="Workspace Name"
      margin="dense"
      required={false}
      select={false}
      type="email"
    />
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
    >
      Cancel
    </WithStyles(Button)>
    <WithStyles(Button)
      color="primary"
      disabled={true}
    >
      Export
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</WithStyles(Dialog)>
`;
